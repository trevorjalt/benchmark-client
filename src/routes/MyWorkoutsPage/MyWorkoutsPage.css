.MyWorkoutsPage {
    display: flex;
    align-items: center;
    flex-direction: column;
    justify-content: center;
}

.MyWorkouts__title {
    margin-bottom: 7vh;
    font-size: 4.5vh;
}