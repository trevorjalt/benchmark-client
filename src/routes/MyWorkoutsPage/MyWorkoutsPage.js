import React, { Component } from 'react'
import WorkoutContext from '../../contexts/WorkoutContext'
import WorkoutApiService from '../../services/workout-api-service'
import { Section } from '../../components/Utils/Utils'
import Workout from '../../components/Workout/Workout'


export default class MyWorkoutsPage extends Component {
    static contextType = WorkoutContext

    componentDidMount() {
        this.context.clearError()
        WorkoutApiService.getWorkouts()
            .then(this.context.setWorkoutList)
            .catch(this.context.setError)
    }

    renderWorkouts() {
        const { workoutList = [] } = this.context
        return workoutList.map(workout =>
            <Workout 
                key={workout.id}
                workout={workout}
            />       
        )
    }

    render() {
        const { error } = this.context
        return (
            <div>
                <h1>myWorkouts</h1>
                <Section list className='MyWorkoutsPage'>
                    {error
                    ? <p className='red'>Whoops! There was an error</p>
                    : this.renderWorkouts()}
                </Section>
            </div>
        )
    }
}