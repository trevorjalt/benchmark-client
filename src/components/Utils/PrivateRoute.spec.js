import React from 'react'
import PrivateRoute from './PrivateRoute'
import MyWorkoutsPage from '../../Routes/MyWorkoutsPage/MyWorkoutsPage'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe(`PrivateRoute Component`, () => {
    const props = {
        component: <MyWorkoutsPage />
    }
    
    it (`renders private route`, () => {
        const wrapper = 
            shallow (
                <PrivateRoute {...props} />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})