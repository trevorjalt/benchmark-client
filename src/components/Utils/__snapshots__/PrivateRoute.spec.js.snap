// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute Component renders private route 1`] = `
<Route
  render={[Function]}
/>
`;
